cmd/charm/charmcmd/push.go

```diff
@@ -239,179 +236,133 @@ func validateResources(resources map[string]string, meta *charm.Meta) error {
 	}
 }
 
-type LogEntry struct {
-	Commit  string
-	Name    string
-	Email   string
-	Date    time.Time
-	Message string
+func updateExtraInfo(id *charm.URL, srcDir string, client *csClient) error {
+	info := getExtraInfo(srcDir)
+	if info == nil {
+		return nil
```

Comment:
I don't understand why git doesn't want to show the two tabs of indentation here. They're definitely in the source AFAICS.